src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/CreateFinancialTransactionCategoryIT.java

```diff
@@ -0,0 +1,221 @@
+                        .content(objectMapper.writeValueAsString(financialTransactionCategoryCreateDTO))
+                        .with(user("1"))
+                        .with(csrf()))
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()));
+
+        assertEquals(0, financialTransactionCategoryRepository.count());
+        assertEquals(0, walletRepository.count());
+        assertEquals(1, userRepository.count());
+    }
+
+    //invalid type
+
+    @DisplayName("Should return error when type is invalid")
+    @Test
+    void testCreateInvalidType() throws Exception {
+        var financialTransactionCategoryCreateDTO = new FinancialTransactionCategoryCreateDTO
+                ("Category",
+                        FinancialTransactionType.valueOf("INVALID"),
+                        1L);
+
+        mockMvc.perform(post("/api/categories")
+                        .contentType(APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(financialTransactionCategoryCreateDTO))
+                        .with(user("1"))
+                        .with(csrf()))
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()));
+
+        assertEquals(0, financialTransactionCategoryRepository.count());
+        assertEquals(0, walletRepository.count());
+        assertEquals(1, userRepository.count());
+    }
+
+    private User createTestUser() {
+        final User userOne = User.builder()
+                .id(1L)
+                .userName("userone")
+                .email("Email@wp.pl")
+                .password("Password1@")
+                .userStatus(UserStatus.VERIFIED)
+                .build();
+        return userRepository.save(userOne);
+    }
+
+
+    private FinancialTransactionCategory createTestFinancialTransactionCategory() {
+        final FinancialTransactionCategory testFinancialTransactionCategory = FinancialTransactionCategory.builder()
+                .name("name")
+                .type(FinancialTransactionType.INCOME)
+                .user(createTestUser())
+                .build();
+        return financialTransactionCategoryRepository.save(testFinancialTransactionCategory);
+    }
+
+        private FinancialTransactionCategoryDTO createTestFinancialTransactionCategoryDTO (){
```

Comment:
Code formatting:
- indentation too large for this line
- the SPACE BAR character after the ')'